spec/vrt/node_spec.rb

```diff
@@ -49,8 +49,41 @@
       expect(mappings.keys).to eq(VRT::MAPPINGS)
     end
 
-    it 'has the right values' do
-      expect(mappings).to include(cvss_v3: 'b')
+    context 'cvss_v3' do
+      it 'has the right values' do
+        expect(mappings).to include(cvss_v3: 'b')
+      end
+    end
+
+    context 'remediation_advice' do
+      let(:id) { 'server_security_misconfiguration.unsafe_cross_origin_resource_sharing' }
+
+      it 'has the expected remediation advice' do
+        expect(mappings[:remediation_advice]).to match hash_including(
+          remediation_advice: 'This is advice'
+        )
+      end
+
+      it 'has the expected (concatenated) references' do
+        expect(mappings[:remediation_advice]).to match hash_including(
+          references: [
+            'https://www.owasp.org/index.php/HTML5_Security_Cheat_Sheet#Cross_Origin_Resource_Sharing',
+            'https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS',
+            'https://www.owasp.org/index.php/Top_10_2013-A5-Security_Misconfiguration',
+            'http://projects.webappsec.org/w/page/13246959/Server%20Misconfiguration'
+          ]
+        )
+      end
+    end
+
+    context 'cwe' do
+      it 'has the exepected (concatenated) CWE IDs' do
+        expect(mappings[:cwe]).to eq [
+          'CWE-942',
+          'CWE-933',
+          'CWE-2000'
+        ]
+      end
     end
   end
```

Comment:
Even though this is all tested in the `node_spec` I think it might more clearly document the functionality to have a similar spec in the `mapping_spec` like:

```ruby
context 'mapping with multiple keys' do
  let(:advice) { described_class.new(:remediation_advice) }
  let(:version) { '999.999' }

  subject { advice.get(id_list, version) }

  context 'when only 1 key is present' do
    let(:id_list) { %i[server_security_misconfiguration] }

    it 'returns both keys' do
      is_expected.to match a_hash_including(
        remediation_advice: nil,
        references: [
          'https://www.owasp.org/index.php/Top_10_2013-A5-Security_Misconfiguration',
          'http://projects.webappsec.org/w/page/13246959/Server%20Misconfiguration'
        ]
      )
    end
  end

  context 'with nested arrays' do
    let(:id_list) { %i[server_security_misconfiguration unsafe_cross_origin_resource_sharing] }

    it 'merges with parent items in the correct order' do
      is_expected.to match a_hash_including(
        references: [
          'https://www.owasp.org/index.php/HTML5_Security_Cheat_Sheet#Cross_Origin_Resource_Sharing',
          'https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS',
          'https://www.owasp.org/index.php/Top_10_2013-A5-Security_Misconfiguration',
          'http://projects.webappsec.org/w/page/13246959/Server%20Misconfiguration'
        ]
      )
    end
 end
```

Reply:
maybe more edge cases can be included if you can think of any, but this way these edge cases will not pollute the `node_spec`

Reply:
@adamrdavid good call, added a few tests for this in mapping_spec.